server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/get/TransportGetSnapshotsAction.java

```diff
@@ -438,7 +470,11 @@ private void snapshots(
                 snapshotIdsToIterate,
                 ignoreUnavailable == false,
                 task::isCancelled,
-                (context, snapshotInfo) -> snapshotInfos.add(snapshotInfo),
+                predicate == null ? (context, snapshotInfo) -> snapshotInfos.add(snapshotInfo) : (context, snapshotInfo) -> {
```

Comment:
I wonder if we could use a special predicate object instead of nul to avoid checking for null. We could still have a specific predicate we use to allow the assertions.

Reply:
Slightly similar to what Henning suggested, I wonder if we could fold the logic of checking the nullity of predicates and then test directly into `SnapshotPredicates`, which would provide a `preFilter` and `filter` methods

Reply:
Will do in a follow-up. I have a queued up round of improvements here that really want the special case `null` (where it would be the same code complexity if I used a placeholder predicate and checked for that) :)